src/aosm/azext_aosm/_configuration.py

```diff
@@ -59,9 +62,51 @@ def validate(self):
         if not self.version:
             raise ValidationError("version must be set.")
         if self.blob_sas_url and self.file_path:
-            raise ValidationError("Only one of file_path or blob_sas_url may be set.")
+            raise ValidationError(
+                "Only one of file_path or blob_sas_url may be set."
+            )
         if not (self.blob_sas_url or self.file_path):
-            raise ValidationError("One of file_path or sas_blob_url must be set.")
+            raise ValidationError(
+                "One of file_path or sas_blob_url must be set."
+            )
+
+
+@dataclass
+class VhdArtifactConfig(ArtifactConfig):
+    image_disk_size_GB: Optional[Union[str, int]] = None
```

Comment:
Worth a comment saying if you change or add to these then check whether VHD_EXTRA_PARAMETERS in constants.py needs changing.